src/__lib__/html.js

```diff
@@ -31,16 +31,27 @@ export default ({
         <body>
             <div id='app'>${html}</div>
             <script>
-window.gu = ${JSON.stringify(
-    {
-        app: {
-            data,
-            cssIDs,
-        },
-    },
-    null,
-    2,
-)};
+            window.gu = ${JSON.stringify({
+                app: {
+                    data: {
+                        ...data,
+                        // instead of writing the data in the `content` object,
+                        // write calls to the innerHTML of elements we've already
+                        // rendered it into
+                        content: Object.entries(data.content).reduce(
+                            (content, [key]) => ({
+                                [key]: `document.querySelector('[data-content-${key}]').innerHTML`,
```

Comment:
If the author of a component hasn't put the `data-content-xxx` attribute on the element in the JSX, should we use `data.content.xxx` as the fallback value?

Reply:
i'm not sure  – this is obviously v prone to breaking and so not ideal. i guess it would break in dev if you don't add the necessary keys? i suppose i imagine this to be only for things you never want to re-render

Reply:
i think maybe this shows a poor way to approach the problem, but it helps up get nearer a better one?

not sure that's something we want in master or not...

Reply:
true, it would break in dev without the necessary data attribute, I just wondered if we could guard against it, but perhaps doing that wouldd increase the chances of people forgetting/not realising they should add the attribute.